### ISSUE TITLE
FilteredIterator remove() removes wrong element from underlying Iterator

### ISSUE THREAD

**Issue by Timmeey** [yegor256/cactoos#82]
Under certain circumstances, remove() on a FilteredIterator removes the wrong element from the underlying iterator.

Reproduce:
Create List containing 0,1,2,3. (0,1,2,3)
Retrieve FilteredIterator over this List, set input-> true as filter function (no real filtering)
Retrieve first element (0).
Call hasNext() on FilteredIterator
call remove() on FilteredIterator
Expected: Check original List, it should not contain 0 anymore, but still have 1 in it. (1,2,3)
Actual: List does not contain 1 anymore (0,2,3)

I forked this repo and wrote a quick [unit test](https://github.com/Timmeey/cactoos/commit/976f762454d3a15a7bf6491a2ae1ae259e6e5bf8) to showcase this bug.
The unittest unfortunately does not follow [single statement unit test](http://www.yegor256.com/2017/05/17/single-statement-unit-tests.html) since i don't know how to reproduce a multi-stage bug within a single statement.

What happens here is [remove() Removes from the underlying collection the last element returned by this iterator (optional operation).](https://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html)
Since hasNext() on the FilteredIterator already calls next() on the underlying iterator. Because of that, the next-pointers" in the underlying and the filtered iterator diverge.
Therefore a subsequent remove() will call remove() on the underlying iterator, which already has returned the next element. 
But since the next() was not called yet on the filteredIterator this behaviour is unexpected.

I tried to come up with a solution, but since the filter operation needs to look ahead in the underlying operator, i cannot see a way to offer the remove() operation. But luckily the remove() operation is an optional operation, we can just remove it. For that is opened a PR.

Tl;dR
FilteredIterator next pointer diverge from the underlying Iterators next pointer. Therefore the remove-method does not work as expected.
UnitTest [here](https://github.com/Timmeey/cactoos/commit/976f762454d3a15a7bf6491a2ae1ae259e6e5bf8)

PR which makes remove() a not supported operation here #83

**Comment by 0crat** [comment]
@timmeey I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

**Comment by yegor256** [comment]
@Timmeey thanks for the report. I believe it's fixed in #83

**Comment by Timmeey** [comment]
Correct, my PR #83 closed this issue.
